src/policy/v3_policy.cpp

```diff
@@ -214,10 +213,27 @@ std::optional<std::pair<std::string, CTransactionRef>> SingleV3Checks(const CTra
             std::any_of(children.cbegin(), children.cend(),
                 [&direct_conflicts](const CTxMemPoolEntry& child){return direct_conflicts.count(child.GetTx().GetHash()) > 0;});
         if (parent_entry->GetCountWithDescendants() + 1 > V3_DESCENDANT_LIMIT && !child_will_be_replaced) {
-            return std::make_pair(strprintf("tx %u (wtxid=%s) would exceed descendant count limit",
-                             parent_entry->GetSharedTx()->GetHash().ToString(),
-                             parent_entry->GetSharedTx()->GetWitnessHash().ToString()),
-                nullptr);
+            // Allow sibling eviction for v3 transaction: if another child already exists, even if
+            // we don't conflict with it, consider evicting it under RBF rules. We rely on v3 rules
+            // only permitting 1 descendant, as otherwise we would need to have logic for deciding
+            // which descendant to evict. Skip if this isn't true, e.g. if this transaction has
+            // extra descendants due to a reorg.
+            const bool consider_sibling_eviction{children.size() == 1};
```

Comment:
Should we also exclude the case where the sibling transaction has descendants of its own, which can happen as the result of a reorg? My concern is that our RBF rules today don't enforce incentive compatibility in such a situation, so it's not clear that we should support this case for now (post-cluster-mempool, this would be fine).

Reply:
Changed this to be `parent_entry->GetCountWithDescendants() == 2` so no grandchildren and no other siblings.